code/__DEFINES/directions.dm

```diff
@@ -31,3 +31,21 @@
 
 /// Inverse direction, taking into account UP|DOWN if necessary.
 #define REVERSE_DIR(dir) ( ((dir & 85) << 1) | ((dir & 170) >> 1) )
+/// returns TRUE if the direction is EAST or WEST
+#define DIR_JUST_HORIZONTAL(dir) (dir & (EAST | WEST))
+/// returns TRUE if the direction is NORTH or SOUTH
+#define DIR_JUST_VERTICAL(dir) (dir & (NORTH | SOUTH))
+/*
+* For getting coordinate signs from a direction define. I.E. NORTHWEST is (-1,1), SOUTH is (0,-1)
+* Returns a length 2 list where the first value is the sign of x, and the second is the sign of y
+*/
+/proc/get_signs_from_direction(direction)
+	var/x_sign = 1
+	var/y_sign = 1
+	x_sign = ((direction & EAST) ? 1 : -1)
+	y_sign = ((direction & NORTH) ? 1 : -1)
+	if(DIR_JUST_VERTICAL(direction))
```

Comment:
This will be true if the direction contains ANY vertical direction at all. So NE is true, SW is true, S is true, E is false